kyaml/kio/byteio_writer.go

```diff
@@ -50,7 +50,12 @@ type ByteWriter struct {
 
 var _ Writer = ByteWriter{}
 
-func (w ByteWriter) Write(nodes []*yaml.RNode) error {
+func (w ByteWriter) Write(inputNodes []*yaml.RNode) error {
+	// Copy the nodes to prevent writer from mutating the original nodes.
+	var nodes []*yaml.RNode
```

Comment:
`nodes = make([]*yaml.RNode, len(inputNodes)`


Actually, just to be clear, pls extract to a small function.
No need for append since length is known.

```
func copyRNodes(in []*yaml.RNode) (s []*yaml.RNode) {
 s = make([]*yaml.RNode, len(in))
 for i range in {
   s[i] = in[i].Copy()
 }
 return
}
```